lib/tls.js

```diff
@@ -189,8 +183,7 @@ exports.checkServerIdentity = function checkServerIdentity(host, cert) {
 
   if (!valid) {
     var err = new Error(
-        util.format('Hostname/IP doesn\'t match certificate\'s altnames: %j',
-                    reason));
+        `Hostname/IP doesn't match certificate's altnames: "${reason}"`);
```

Comment:
Not well. 
```javascript
var m = {};
m.n = m;
console.log(`${m}`);
```
Outputs: "`[object Object]`"

Looking this over, I think it's safer to go ahead and do a variation on the `JSON.stringify` performed by the original version:

```
function stringify(val) {
  try {
    return JSON.stringify(val);
  } catch(_) {
    return '[Circular]';
  }
};
// ...
// ...
`Hostname/IP doesn't match certificate's altnames: ${stringify(reason)}`;
```